go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -91,3 +91,68 @@ type Plugin interface {
 	// Properties desired by the plugin
 	GetProperties() PluginProperties
 }
+
+type PluginAbortOverride interface {
+	OnAbort(ctx context.Context, tCtx pluginsCore.TaskExecutionContext, resource client.Object) (behavior AbortBehavior, err error)
+}
+
+type AbortBehavior struct {
+	// Optional override to the default k8s Resource being acted on.
+	Resource       client.Object
+	DeleteResource bool
+	Update         *UpdateResourceOperation
+	Patch          *PatchResourceOperation
+	// Determines whether to delete the Resource if the specified operations return an error.
+	DeleteOnErr    bool
+}
+
+type PatchResourceOperation struct {
+	Patch     client.Patch
+	Options []client.PatchOption
+}
+
+type UpdateResourceOperation struct {
+	Options []client.UpdateOption
+}
+
+func AbortBehaviorPatchDefaultResource(patchOperation PatchResourceOperation, deleteOnErr bool) AbortBehavior {
+	return AbortBehaviorPatch(patchOperation, deleteOnErr, nil)
+}
+
+func AbortBehaviorPatch(patchOperation PatchResourceOperation, deleteOnErr bool, resource client.Object) AbortBehavior {
+	return AbortBehavior {
+		Resource: resource,
+		DeleteResource: false,
+		Update: nil,
+		Patch: &patchOperation,
+		DeleteOnErr: deleteOnErr,
+	}
+}
+
+func AbortBehaviorUpdateDefaultResource(updateOperation UpdateResourceOperation, deleteOnErr bool) AbortBehavior {
+	return AbortBehaviorUpdate(updateOperation, deleteOnErr, nil)
+}
+
+func AbortBehaviorUpdate(updateOperation UpdateResourceOperation, deleteOnErr bool, resource client.Object) AbortBehavior {
+	return AbortBehavior {
+		Resource: resource,
+		DeleteResource: false,
+		Update: &updateOperation,
+		Patch: nil,
+		DeleteOnErr: deleteOnErr,
+	}
+}
+
+func AbortBehaviorDeleteDefaultResource() AbortBehavior {
+	return AbortBehaviorDelete(nil)
+}
+
+func AbortBehaviorDelete(resource client.Object) AbortBehavior {
+	return AbortBehavior {
+		Resource: resource,
+		DeleteResource: true,
+		Update: nil,
```

Comment:
avoid setting zero values
pointers/refs have zero value of nil and bools have zero value of false. I appreciate that the zero behaviour is the default in all cases :), lets drop them and make them clean.